src/ast/passes/simplify_types.cpp

```diff
@@ -90,12 +91,26 @@ std::optional<Expression> SimplifyTypes::visit(Binop &op)
   // strict about the inner types of the tuple matching up due to size
   // and alignment issus.
   // https://github.com/bpftrace/bpftrace/pull/4523#discussion_r2382109017
+  var_id_++;
+  auto *left_tuple_var = ast_.make_node<Variable>("$$binop_tuple_left_" +
+                                                      std::to_string(var_id_),
+                                                  Location(op.left.loc()));
+  auto *right_tuple_var = ast_.make_node<Variable>("$$binop_tuple_right_" +
+                                                       std::to_string(var_id_),
+                                                   Location(op.right.loc()));
+  auto *left_var_assign = ast_.make_node<AssignVarStatement>(
+      left_tuple_var, clone(ast_, op.left, op.left.loc()), Location(op.loc));
+  auto *right_var_assign = ast_.make_node<AssignVarStatement>(
+      right_tuple_var, clone(ast_, op.right, op.right.loc()), Location(op.loc));
```

Comment:
I think you want to do the following:
- If both sides are anonymous tuples, e.g. `(...) == (...)` then you can fold directly into your expanded expression.
- Otherwise, you need to lazily bound whatever side is *not* already a variable and/or map into that. So `$x == (...)` because `$x == $$binop_tuple_right_`, `$x == $y` is still `$x == $y` and `$x == @y` is still `$x == @y`